### PR TITLE
Allow configuring token in java sdk

### PR DESCRIPTION
**Pull request by eddiechayes**

### CODE REVIEW

File: java/src/main/java/com/konfigthis/splitit/client/ApiClient.java [hunk 1]
```diff
@@ -65,6 +65,7 @@
 public class ApiClient extends ApiClientCustom {
 
     private String basePath = "https://web-api-v3.production.splitit.com";
+    private String tokenPath = "https://id.production.splitit.com/connect/token";
```
Comment (dphuang2) [inline java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: I think `tokenUrl` makes more sense.

Comment (dphuang2) [reply java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: but actually, this is an internal detail. So no need to change.

File: java/src/main/java/com/konfigthis/splitit/client/ApiClient.java [hunk 2]
```diff
@@ -240,6 +240,30 @@ public ApiClient setBasePath(String basePath) {
...
+     * @return Token path
+     */
+    public String getTokenPath() {
+        return tokenPath;
+    }
+
+    /**
+     * Set token path
+     *
+     * @param tokenPath Token URL for oauth (e.g https://id.production.splitit.com/connect/token)
```
Comment (dphuang2) [inline java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: this comment might be hard to add to template.

**Review by dphuang2 — APPROVED** [review]

File: java/src/main/java/com/konfigthis/splitit/client/ApiClient.java [hunk 3]
```diff
@@ -165,18 +167,16 @@ public ApiClient(String basePath, String clientId, String clientSecret, Map<Stri
         if (basePath != null) {
             this.basePath = basePath;
         }
-
-        String tokenUrl = "https://id.production.splitit.com/connect/token";
-        if (!"".equals(tokenUrl) && !URI.create(tokenUrl).isAbsolute()) {
+        if (!"".equals(tokenPath) && !URI.create(tokenPath).isAbsolute()) {
             URI uri = URI.create(getBasePath());
-            tokenUrl = uri.getScheme() + ":" +
+            tokenPath = uri.getScheme() + ":" +
```
Comment (eddiechayes) [inline java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: is this right actually? I dont want to overwrite the instance variable do I?

File: java/src/main/java/com/konfigthis/splitit/client/ApiClient.java (hunk 2 above)
Comment (eddiechayes) [reply java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: this is already generated from the template 😎

File: java/src/main/java/com/konfigthis/splitit/client/ApiClient.java (hunk 3 above)
Comment (eddiechayes) [reply java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: oh no its fine, after the scheme gets added once then it wont anymore

Comment (dphuang2) [reply java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: when is this constructor even used?

File: java/src/main/java/com/konfigthis/splitit/client/ApiClient.java (hunk 2 above)
Comment (dphuang2) [reply java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: oh i see, yeah its fine then.

File: java/src/main/java/com/konfigthis/splitit/client/ApiClient.java (hunk 3 above)
Comment (eddiechayes) [reply java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: Shouldnt every constructor have to go through this one ultimately in order to set up oauth?

Comment (eddiechayes) [reply java/src/main/java/com/konfigthis/splitit/client/ApiClient.java]: actually, I'm a little confused. there's one other constructor that calls `authentications.put(...)`, but it doesn't reference the tokenUrl at all?

**Pull request merged**